### PR TITLE
LF-50: Integrate LiFi Contract with nxtp on staging

### PR DESCRIPTION
**Pull request by maxklenk**
When activated by specifying supported chains in `REACT_APP_LIFI_CONTRACT_ENABLED_CHAINS_JSON`, the returned routes will be checked if they can be executed using our contract. The user then only has to submit a single transaction to perform swaps and a nxtp based transfer.
Screenshot 2021-10-23 at 8 02 51 PM
Screenshot 2021-10-23 at 8 07 07 PM
.

### CODE REVIEW

File: src/components/Swap.tsx [hunk 1]
```diff
@@ -66,12 +67,14 @@ const Swap = ({
 
   // Options
   const [optionSlippage, setOptionSlippage] = useState<number>(3)
+  // const [optionInfiniteApproval, setOptionInfiniteApproval] = useState<boolean>(true)
+  const [optionUseLifiContract, setOptionUseLifiContract] = useState<boolean>(true)
 
   // Routes
-  const [routes, setRoutes] = useState<Array<Array<TransferStep>>>([])
+  const [routes, setRoutes] = useState<Array<Array<Step>>>([])
```
Comment (mathiasmoeller) [inline src/components/Swap.tsx:74]: Shouldn't we maybe use routes here?
```suggestion
  const [routes, setRoutes] = useState<Array<Route>>([])
```

File: src/components/Swap.tsx [hunk 2]
```diff
@@ -66,12 +67,14 @@ const Swap = ({
...
   const [routesLoading, setRoutesLoading] = useState<boolean>(false)
   const [noRoutesAvailable, setNoRoutesAvailable] = useState<boolean>(false)
-  const [selectedRoute, setselectedRoute] = useState<Array<TransferStep>>([])
+  const [selectedRoute, setselectedRoute] = useState<Array<Step>>([])
```
Comment (mathiasmoeller) [inline src/components/Swap.tsx:77]: And here
```suggestion
  const [selectedRoute, setselectedRoute] = useState<Route>([])
```

File: src/components/Swap.tsx [hunk 3]
```diff
@@ -209,7 +212,8 @@ const Swap = ({
         try {
           const result = await axios.post<any>(process.env.REACT_APP_API_URL + 'transfer', { deposit, withdraw }, config)
           // filter if needed
-          const routes: Array<Array<TransferStep>> = result.data
+          const routesRaw: Array<Array<Step>> = result.data
```
Comment (mathiasmoeller) [inline src/components/Swap.tsx:215]: I mean everywhere where we have a list of steps 😁

File: src/components/Swapping.tsx [hunk 4]
```diff
@@ -314,49 +317,37 @@ const Swapping = ({ route, updateRoute, onSwapDone }: SwappingProps) => {
     setIsSwapping(true)
   }
 
-  const getRecevingInfo = (step: TransferStep) => {
-    let toChain: Chain, toToken: Token;
-    switch(step.action.type) {
+  const getRecevingInfo = (step: Step) => {
+    let toChain: Chain, toToken: Token
+    switch (step.action.type) {
```
Comment (mathiasmoeller) [inline src/components/Swapping.tsx:322]: This is missing the default case now, isn't it?

File: src/services/1Inch.ts [hunk 5]
```diff
@@ -93,6 +93,8 @@ const getTransaction = async (chainId: number, fromTokenAddress: string, toToken
   const path: Array<any> = result.data.protocols ? result.data.protocols[0].map((step: Array<any>) => step[0]) : []
   const tx = result.data.tx
 
+  console.info('1inch data', result.data)
```
Comment (mathiasmoeller) [inline src/services/1Inch.ts:96]: ✂️

File: src/services/nxtp.tsx [hunk 6]
```diff
@@ -75,7 +75,7 @@ export const getTransferQuote = async (
   initiator?: string,
 ): Promise<AuctionResponse | undefined> => {
   // Create txid
-  const transactionId = getRandomBytes32();
+  const transactionId = getRandomBytes32()
```
Comment (mathiasmoeller) [inline src/services/nxtp.tsx:78]: Maybe we use that for our step ids as well?

Comment (maxklenk) [reply src/services/nxtp.tsx:78]: Yes, that would be good.

File: src/components/Swap.tsx (hunk 1 above)
Comment (maxklenk) [reply src/components/Swap.tsx:74]: This PR has been build before we implemented Routes.

**Comment by maxklenk** [comment]
Merged in new implementation

**Pull request closed without merging**